src/carlib/sports_car.rs

```diff
@@ -1,63 +1,40 @@
 use crate::carlib::car::Car;
-use crate::carlib::car_concept::CarConcept;
-use crate::carlib::car_impl::CarImpl;
+
+use super::car::UNNAMED_CAR;
 
 // Rust seems to encourage composition over direct inheritance.
 // I like that. 👍
+const DEFAULT_ACCELERATION: f64 = 0.84; // I like fast cars and .84 is double of .42.
 pub struct Sportscar {
-    // A Sportcar is a class that has an implementation
-    // `imp` of a car called `CarImpl`. Makes sense.
-    imp: CarImpl, //Perhaps there is a better name than imp?
+    pub acceleration: f64,
+    pub name: String,
+    pub speed: f64,
 }
 
 // Here is the implementation of my sportscar.
 impl Sportscar {
-    pub fn new() -> Sportscar {
-        let mut obj = Sportscar {
-            imp: CarImpl::new(),
-        };
-        obj.imp.acceleration = 0.84; // I like fast cars and .84 is double of .42.
-        obj
-    }
-    pub fn new_named(name: String) -> Sportscar {
-        let mut obj = Sportscar {
-            imp: CarImpl::new_named(name),
-        };
-        // I don't quite like the duplication here.
-        obj.imp.acceleration = 0.84;
-        obj
+    pub fn new(name: Option<String>, acceleration: Option<f64>) -> Self {
+        Self {
+            name: name.unwrap_or(UNNAMED_CAR.to_string()),
+            acceleration: acceleration.unwrap_or(DEFAULT_ACCELERATION),
+            speed: 0.0,
+        }
```

Comment:
The `Option<String>` makes it much better as it eliminates the need for the second ctor. (perhaps a reference would be even better?)